src/menu/views/cpakfs_manager.c

```diff
@@ -0,0 +1,1016 @@
+    }
+};
+
+static void populate_list_cpakfs() {  
+    if (has_mem && !ctr_p_data_loop) {
+        free_controller_pak_name_notes();
+        //TODO: enable actions when a cpak is present
+        if (dir_findfirst(CPAK_MOUNT_ARRAY[controller_selected], &dir_entry) >= 0) {
+            char filename_cpak[256];
+            sprintf(filename_cpak, "%s%s", CPAK_MOUNT_ARRAY[controller_selected], dir_entry.d_name);
+            int size = get_block_size_from_fs_path(filename_cpak);
+            if (size < 0) {
+                sprintf(controller_pak_name_notes_bank_size[0], " ");
+            } else {
+                sprintf(controller_pak_name_notes_bank_size[0], "(%-3.3d blocks)", size);
+            }
+            snprintf(controller_pak_name_notes[0], MAX_STRING_LENGTH, "%s", dir_entry.d_name);
+            parse_cpakfs_fullname(dir_entry.d_name, &cpakfs_path_strings[0]);
+            int i = 1;     
+            while(dir_findnext(CPAK_MOUNT_ARRAY[controller_selected], &dir_entry) == 0) {
+                sprintf(filename_cpak, " ");
+                sprintf(filename_cpak, "%s%s", CPAK_MOUNT_ARRAY[controller_selected], dir_entry.d_name);
+                size = get_block_size_from_fs_path(filename_cpak);
+                if (size < 0) {
+                    sprintf(controller_pak_name_notes_bank_size[i], " ");
+                } else {
+                    sprintf(controller_pak_name_notes_bank_size[i], "(%-3.3d blocks)", size);
+                }
+                snprintf(controller_pak_name_notes[i], MAX_STRING_LENGTH, "%s", dir_entry.d_name);
+
+                parse_cpakfs_fullname(dir_entry.d_name, &cpakfs_path_strings[i]);
+            
+                i++;
+                if (i >= MAX_NUM_NOTES) break;
+
+                ctr_p_data_loop = true;
+            }
+        }
+    }
+}
+
+static void dump_complete_cpak(int _port) {
+    sprintf(failure_message_note, " ");
+    uint8_t* data = malloc(MEMPAK_BLOCK_SIZE * 128 * sizeof(uint8_t));
+
+    if (!data) {
+        //"Memory allocation failed!"
+        sprintf(failure_message_note, "Memory allocation failed!");
+        error_message_displayed = true;
+        return;
+    }
+    
+    surface_t *d;
+    for (int i = 0; i < 128; i++) {
+
+        d = display_try_get();
+        rdpq_attach(d, NULL);
+
+        ui_components_layout_draw();
+
+        
+        ui_components_messagebox_draw(
+            "Do you want to dump the CPAK?\n\n"
+            "A: Yes     B: No"
+        );   
+        
+
+        if (read_mempak_sector(_port, i, data + (i * MEMPAK_BLOCK_SIZE)) != 0) {
+            //debugf("Failed to read mempak sector %d\n", i);
+            free(data);
+            sprintf(failure_message_note, "Failed to read mempak sector %d", i);
+            error_message_displayed = true;
+            return;
+        }
+
+        ui_components_loader_draw((float) i / 128.0f, "Dumping Controller Pak...");
+
+        rdpq_detach_show();
+    }
+
+    get_rtc_time(string_datetime_cpak);
```

Comment:
We should somehow try to align closer with ISO8601 when saving. i.e. rather than `CPAK_2025.09.13_21h59m28s.mpk` it might be preferable to be ` `CPAK_2025-09-13_215928.mpk` ( but, is only a thought, and potentially requires an update from the libdragon SDK).